### ISSUE TITLE
Delegate parameter nullability attributes not respected in lambda analysis

### ISSUE THREAD

**Issue by RikkiGibson** [dotnet/roslyn#45571]
**Version Used**: 83e13e1

**Steps to Reproduce**:

```cs
#nullable enable
using System.Diagnostics.CodeAnalysis;

delegate void MyDel<T>([AllowNull] T t);

class C
{
    static void F<T>(T t, MyDel<T> f)
    {
        f(default);
    }
    static void Main()
    {
        F(new object(), y => y.ToString());
    }
}
```

[SharpLab repro](https://sharplab.io/#v2:EYLgtghgzgLgpgJwD4GIB2BXANliwtwAEcaeBAsAFAACADIdQIwB0AIgJYQDmaA9rOwDGUZgGFeAEzgBBUlgCeUdlADcVKlIJcI8BgBZCAWXms4WADwAVAHwAKANrScvAO4A5bFgC6hS4RgAlGqUVNQATISiVADeVITxDIwAbPqEAGJWdn4wADRGJmaZhABmAXEJsZQJ1SW2UsUQ2IHB1QC+5fFMKdQGhhDsaLZlVRUdNWm2aHAuhLzAAFZwgjBDefKEALzWhPLMlrwAyjAIA1xDQWPtlK1AA===) 

**Expected Behavior**: NullReferenceReceiver warning on `y.ToString()`

**Actual Behavior**: NRE at runtime

**Comment by RikkiGibson** [comment]
Another scenario which is likely a corollary:

```cs
#nullable enable
using System.Diagnostics.CodeAnalysis;

public delegate T Del<T>([DisallowNull] object? obj);

public class C {
    public T M<T>(Del<T> del)
    {
        return del(new object());
    }
    
    public void M1()
    {
        M(obj => obj).ToString(); // this warns, but it shouldn't
    }
}
```

**Comment by RikkiGibson** [comment]
Very likely solved by #46953

**Comment by RikkiGibson** [comment]
It's not :smile:

**Comment by jcouv** [comment]
From our discussion, the current behavior (no dereference warning) is by-design.
However as part of adding support for attributes on lambdas we're adding checks of nullability attributes on those conversions and there should be conversion warnings in those scenarios. This is tracked by https://github.com/dotnet/roslyn/issues/56668
